examples/plot_model_agnostic_importance.py

```diff
@@ -0,0 +1,166 @@
+# We then compute the p-values using LOCO
+# with a linear, and then a non-linear model. When using a
+# misspecified model, such as a linear model for this dataset, LOCO fails to reject the null
+# similarly to d0CRT. However, when using a non-linear model (SVC), LOCO is able to
+# identify the important variables.
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+non_linear_model = SVC(kernel="rbf", random_state=0)
+linear_model = LogisticRegressionCV(Cs=np.logspace(-3, 3, 5))
+
+importances_linear = []
+importances_non_linear = []
+for train, test in cv.split(X):
+    non_linear_model_ = clone(non_linear_model)
+    linear_model_ = clone(linear_model)
+    non_linear_model_.fit(X[train], y[train])
+    linear_model_.fit(X[train], y[train])
+
+    vim_linear = LOCO(
+        estimator=linear_model_, loss=log_loss, method="predict_proba", n_jobs=2
+    )
+    vim_non_linear = LOCO(
+        estimator=non_linear_model_,
+        loss=hinge_loss,
+        method="decision_function",
+        n_jobs=2,
+    )
+    vim_linear.fit(X[train], y[train])
+    vim_non_linear.fit(X[train], y[train])
+
+    importances_linear.append(vim_linear.importance(X[test], y[test])["importance"])
+    importances_non_linear.append(
+        vim_non_linear.importance(X[test], y[test])["importance"]
+    )
+
+
+################################################################################
+# To select variables using LOCO, we compute the p-values using a t-test over the
+# importance scores.
+
+_, pval_linear = ttest_1samp(importances_linear, 0, axis=0, alternative="greater")
+_, pval_non_linear = ttest_1samp(
+    importances_non_linear, 0, axis=0, alternative="greater"
+)
+
+df_pval = pd.DataFrame(
+    {
+        "pval": np.hstack([pval_dcrt, pval_linear, pval_non_linear]),
+        "method": ["d0CRT"] * 2 + ["LOCO-linear"] * 2 + ["LOCO-non-linear"] * 2,
+        "Feature": ["X1", "X2"] * 3,
+    }
+)
+df_pval["minus_log10_pval"] = -np.log10(df_pval["pval"])
+
+
+#################################################################################
+# Plot the :math:`-log_{10}(pval)` for each method and variable
+# ---------------------------------------------------------------------------
+fig, ax = plt.subplots()
+sns.barplot(
+    data=df_pval,
+    y="Feature",
+    x="minus_log10_pval",
+    hue="method",
+    palette="muted",
+    ax=ax,
+)
+ax.set_xlabel("-$\\log_{10}(pval)$")
+ax.axvline(
+    -np.log10(0.05), color="k", lw=3, linestyle="--", label="-$\\log_{10}(0.05)$"
+)
+ax.legend()
+plt.show()
+
+
+#################################################################################
+# As expected, when using linear models (d0CRT and LOCO-linear) that are misspecified,
+# the varibles are not selected. This highlights the benefit of using model-agnostic
+# methods such as LOCO, which allows for the use of models that are expressive enough
+# to capture the underlying data generating process.
```

Comment:
I don't like the expression `data generating process` because we want to explain the data and not the generation of data.
Can you change it, please?

Reply:
This phrasing comes from [Verdinelli et al.](https://arxiv.org/abs/2303.05981)

> Population level feature importance is concerned with measuring how important a particular feature is in the underlying data generating process

I think that they also mean "explain the data".

Reply:
Yes, you are right. It's just that, in the context of experimental data, we often want to understand the underlying process which generates the data. However, when the data are synthetic, it can be a bit confusing to talk about the "data generating process".

Reply:
I see your point. I changed the formulation.